src/main/java/ari/parser/Parser.java

```diff
@@ -0,0 +1,295 @@
+     *
+     * @param command user input: command and description
+     * @return command to execute
+     */
+    public Command parse(String command) {
+        String[] cmdArray = command.split(" ");
+
+        String commandWord = cmdArray[0].toLowerCase();
+
+        switch (commandWord) {
+        case ByeCommand.COMMAND_WORD:
+            return new ByeCommand();
+        case ListCommand.COMMAND_WORD:
+            return new ListCommand();
+        case MarkCommand.COMMAND_WORD:
+            return prepareMark(command);
+        case UnmarkCommand.COMMAND_WORD:
+            return prepareUnmark(command);
+        case TodoCommand.COMMAND_WORD:
+            return prepareTodo(command);
+        case DeadlineCommand.COMMAND_WORD:
+            return prepareDeadline(command);
+        case EventCommand.COMMAND_WORD:
+            return prepareEvent(command);
+        case DeleteCommand.COMMAND_WORD:
+            return prepareDelete(command);
+        case FindCommand.COMMAND_WORD:
+            return prepareFind(command);
+        default:
+            return new IncorrectCommand("I am sorry Master, I am afraid I do not know what you mean");
+        }
+    }
+
+    /**
+     * Assigns methods to execute from input of save file if it exists
+     *
+     * @param instruction command to run
+     * @param description description of task
+     * @return command to execute
+     */
+    public Command fileParse(String instruction, String description) {
+        String command = instruction + " " + description;
+
+        switch (instruction) {
+        case MarkCommand.COMMAND_WORD:
+            return prepareMark(command);
+        case TodoCommand.COMMAND_WORD:
+            return prepareTodo(command);
+        case DeadlineCommand.COMMAND_WORD:
+            return prepareDeadline(command);
+        case EventCommand.COMMAND_WORD:
+            return prepareEvent(command);
+        default:
+            return new IncorrectCommand("Unknown commands in file!");
+        }
+    }
+
+    /**
+     * Splits file command
+     *
+     * @param fileCommand input from file
+     * @return an array containing the command, whether to mark the task and task description
+     */
+    public String[] fileBreakdown(String fileCommand) {
+        String[] words = fileCommand.split(" ");
+        String[] returnCommands = new String[3];
+
+        returnCommands[0] = words[0]; // commands
+        returnCommands[1] = words[1]; // mark or not mark
+        returnCommands[2] = String.join(" ", Arrays.copyOfRange(words, 2, words.length)); // description
+
+        return returnCommands;
+    }
+
+    /**
+     * Returns the description part of the command
+     *
+     * @param command user input
+     * @return description part of command
+     * @throws EmptyCommandException if description is empty
+     */
+    private String getArgument(String command) throws EmptyCommandException {
+        String[] cmdArray = command.split(" ");
```

Comment:
```suggestion
        String[] commandArray= command.split(" ");
```
Consider changing "cmd" to "command" as it seems like that is the convention being used.